Libraries/LibC/grp.cpp

```diff
@@ -96,52 +93,69 @@ struct group* getgrnam(const char* name)
     return nullptr;
 }
 
-struct group* getgrent()
+static bool parse_grpdb_entry(const String& line)
 {
-    if (!__grdb_stream)
-        setgrent();
-
-    assert(__grdb_stream);
-    if (feof(__grdb_stream))
-        return nullptr;
-
-next_entry:
-    char buffer[1024];
-    ++__grdb_line_number;
-    char* s = fgets(buffer, sizeof(buffer), __grdb_stream);
-    if (!s)
-        return nullptr;
-    assert(__grdb_stream);
-    if (feof(__grdb_stream))
-        return nullptr;
-    String line(s, Chomp);
-    auto parts = line.split(':', true);
+    auto parts = line.split_view(':', true);
     if (parts.size() != 4) {
-        fprintf(stderr, "getgrent(): Malformed entry on line %u: '%s' has %zu parts\n", __grdb_line_number, line.characters(), parts.size());
-        goto next_entry;
+        fprintf(stderr, "getgrent(): Malformed entry on line %u: '%s' has %zu parts\n", s_line_number, line.characters(), parts.size());
+        return false;
     }
-    auto& e_name = parts[0];
-    auto& e_passwd = parts[1];
-    auto& e_gid_string = parts[2];
-    auto& e_members_string = parts[3];
-    auto e_gid = e_gid_string.to_uint();
-    if (!e_gid.has_value()) {
-        fprintf(stderr, "getgrent(): Malformed GID on line %u\n", __grdb_line_number);
-        goto next_entry;
+
+    s_name = parts[0];
+    s_passwd = parts[1];
+
+    auto& gid_string = parts[2];
+    String members_string = parts[3];
```

Comment:
The whole point of using `split_view()` instead of `split()` (which would have returned a `Vector<` of `String`s`>`) is to only copy those strings I need copied, while keeping others (namely, just `gid_string`, it seems?) non-copied (borrowed from `line`).

That being said, we shouldn't really need `members_string` itself to be a `String` here, if there was a `Vector<String> StringView::split()`. But IIRC there's no such method.

Reply:
> The whole point [...] is to only copy those strings I need copied

Right, I agree with that, I'm just talking about this specific `members_string`.

>  if there was a `Vector<String> StringView::split()`

There's a  `Vector<String> StringView::split_view()`, which is why I was talking about `Vector<StringView>` -> `Vector<String>`

Reply:
Ah, I see